### ISSUE TITLE
flake: test-cmd does not start up correctly sometimes

### ISSUE THREAD

**Issue by smarterclayton** [openshift/origin#17458]
test-cmd is failing to start in some runs, it looks like etcd is either starting from an inconsistent state, or we have a more subtle etcd configuration bug:

```
I1126 19:18:49.759427   21216 master_config.go:442] Initializing cache sizes based on 0MB limit
I1126 19:18:49.759506   21216 master_config.go:506] Using the lease endpoint reconciler with TTL=15s and interval=10s
I1126 19:18:49.759529   21216 storage_factory.go:290] storing { apiServerIPInfo} in v1, reading as __internal from storagebackend.Config{Type:"etcd3", Prefix:"kubernetes.io", ServerList:[]string{"https://172.17.0.2:24001"}, KeyFile:"/tmp/openshift/test-cmd/openshift.local.config/master/master.etcd-client.key", CertFile:"/tmp/openshift/test-cmd/openshift.local.config/master/master.etcd-client.crt", CAFile:"/tmp/openshift/test-cmd/openshift.local.config/master/ca.crt", Quorum:true, Paging:false, DeserializationCacheSize:0, Codec:runtime.Codec(nil), Copier:(*runtime.Scheme)(0xc42048b480), Transformer:value.Transformer(nil)}
I1126 19:18:49.760195   21216 storage_factory.go:290] storing { endpoints} in v1, reading as __internal from storagebackend.Config{Type:"etcd3", Prefix:"kubernetes.io", ServerList:[]string{"https://172.17.0.2:24001"}, KeyFile:"/tmp/openshift/test-cmd/openshift.local.config/master/master.etcd-client.key", CertFile:"/tmp/openshift/test-cmd/openshift.local.config/master/master.etcd-client.crt", CAFile:"/tmp/openshift/test-cmd/openshift.local.config/master/ca.crt", Quorum:true, Paging:false, DeserializationCacheSize:0, Codec:runtime.Codec(nil), Copier:(*runtime.Scheme)(0xc42048b480), Transformer:value.Transformer(nil)}
2017-11-26 19:18:49.760625 I | etcdserver/api/v3rpc: grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 172.17.0.2:24001: getsockopt: connection refused"; Reconnecting to {172.17.0.2:24001 <nil>}
I1126 19:18:49.760888   21216 compact.go:55] compactor already exists for endpoints [https://172.17.0.2:24001]
I1126 19:18:49.760942   21216 start_master.go:522] Starting master on 172.17.0.2:28443 (v3.8.0-alpha.0+72daef5-279)
I1126 19:18:49.760948   21216 start_master.go:523] Public master address is https://172.17.0.2:28443
I1126 19:18:49.760956   21216 start_master.go:530] Using images from "openshift/origin-<component>:latest"
2017-11-26 19:18:49.761000 I | embed: peerTLS: cert = /tmp/openshift/test-cmd/openshift.local.config/master/etcd.server.crt, key = /tmp/openshift/test-cmd/openshift.local.config/master/etcd.server.key, ca = /tmp/openshift/test-cmd/openshift.local.config/master/ca.crt, trusted-ca = , client-cert-auth = true
2017-11-26 19:18:49.761020 I | etcdserver/api/v3rpc: grpc: addrConn.resetTransport failed to create client transport: connection error: desc = "transport: dial tcp 172.17.0.2:24001: getsockopt: connection refused"; Reconnecting to {172.17.0.2:24001 <nil>}
2017-11-26 19:18:49.761576 I | embed: listening for peers on https://172.17.0.2:27001
2017-11-26 19:18:49.761624 I | embed: listening for client requests on 172.17.0.2:24001
E1126 19:18:50.619518   21216 controllers.go:116] Server isn't healthy yet. Waiting a little while.
2017-11-26 19:18:50.762075 W | etcdserver: another etcd process is using "/tmp/openshift/test-cmd/etcd/member/snap/db" and holds the file lock.
2017-11-26 19:18:50.762104 W | etcdserver: waiting for it to exit before starting...
E1126 19:18:51.619548   21216 controllers.go:116] Server isn't healthy yet. Waiting a little while.
E1126 19:18:52.619533   21216 controllers.go:116] Server isn't healthy yet. Waiting a little while.
E1126 19:18:53.619578   21216 controllers.go:116] Server isn't healthy yet. Waiting a little while.
```

There should be no other process holding the etcd lock in this test suite.  The run continues (with etcd partially starting, perhaps?) but bootstrap hooks fail:

```
logging error output: "[+]ping ok\n[+]etcd ok\n[+]poststarthook/generic-apiserver-start-informers ok\n[+]poststarthook/start-apiextensions-informers ok\n[+]poststarthook/start-apiextensions-controllers ok\n[+]poststarthook/quota.openshift.io-clusterquotamapping ok\n[+]poststarthook/project.openshift.io-projectcache ok\n[+]poststarthook/project.openshift.io-projectauthorizationcache ok\n[-]poststarthook/security.openshift.io-bootstrapscc failed: reason withheld\n[-]poststarthook/authorization.openshift.io-ensureSARolesDefault failed: reason withheld\n[-]poststarthook/authorization.openshift.io-ensureopenshift-infra failed: reason withheld\n[-]poststarthook/bootstrap-controller failed: reason withheld\n[-]poststarthook/ca-registration failed: reason withheld\n[+]poststarthook/start-kube-aggregator-informers ok\n[+]poststarthook/apiservice-registration-controller ok\n[+]poststarthook/apiservice-status-available-controller ok\n[+]poststarthook/kube-apiserver-autoregistration ok\n[-]autoregister-completion failed: reason withheld\n[-]poststarthook/template.openshift.io-sharednamespace failed: reason withheld\n[-]poststarthook/authorization.openshift.io-bootstrapclusterroles failed: reason withheld\n[+]poststarthook/openshift.io-AdmissionInit ok\n[+]poststarthook/openshift.io-StartInformers ok\n[-]poststarthook/oauth.openshift.io-EnsureBootstrapOAuthClients failed: reason withheld\nhealthz check failed\n"
```

https://storage.googleapis.com/origin-ci-test/pr-logs/pull/17417/test_pull_request_origin_cmd/6510/artifacts/scripts/test-cmd/logs/openshift.log

**Comment by smarterclayton** [comment]
@openshift/sig-master

**Comment by liggitt** [comment]
Wonder if this is the same issue I'm seeing with `openshift start master` dying with ` error: EOF` during initial startup occasionally. Only seen from clean start, and only occasionally. Hadn't been able to track down the error further (my kingdom for an originating stack trace)

**Comment by ironcladlou** [comment]
I noticed a few possible blocking syscalls in the [boltdb open code path](https://github.com/openshift/origin/blob/master/vendor/github.com/boltdb/bolt/db.go#L150) (e.g. `open` and `msync`) in addition to the flock loop which is calling `fcntl`. If any of these take too long, etcd will log the warning we're seeing (even though the problem wasn't necessarily due to the `fcntl` call).

Actual errors from the syscalls (or some other file operations in the etcd open logic) should cause a panic, and no stack trace seems to be present, and [newBackend](https://github.com/openshift/origin/blob/master/vendor/github.com/coreos/etcd/etcdserver/backend.go#L29) does eventually return. If the problem is in fact related to blocking syscalls, it's not clear yet whether delays in any of these calls ultimately has any corrupting effect.

I did learn that `test-cmd.sh` is mounting a physical host volume into the container used by etcd, which I believe makes the process susceptible to past syscall issues during EBS volume resizing mid-test. 

One area I was scrutinizing was the boltdb [init logic](https://github.com/openshift/origin/blob/master/vendor/github.com/boltdb/bolt/db.go#L343) which concludes with `msync`, but I don't have an etcd snapshot file from the failed run to inspect for corrupted pages.

There's already https://github.com/openshift/origin/pull/16686 open to get `test-cmd.sh` using tmpfs, so maybe if we're lucky this is the EBS issue and it can be resolved by the tmpfs fix.

Or, I'm totally off base and it's something else entirely. I haven't yet been able to reproduce the problem locally.
